src/wiseagents/core.py

```diff
@@ -843,19 +843,26 @@ def __new__(cls, *args, **kwargs):
         '''Create a new instance of the class, setting default values for the instance variables.'''
         obj = super().__new__(cls)
         obj._system_message = None
+        obj._user_message = None
         return obj
-    def __init__(self, description : str, system_message: Optional[str] = None):
+    def __init__(self, description : str, system_message: Optional[str] = None, user_message: Optional[str] = None):
```

Comment:
what about user_messages: Optional[List[str]] = []

Reply:
ignore this comment. Being a str doesn't make sense to be a List.
Maybe we can add instead 2 parameter pre_user_message and post_user_message
Ideally to be used in handle_request and handle_response respectively

Reply:
Added pre_user_message and post_user_message.